### PR TITLE
Implement shape rotations

### PR DESCRIPTION
**Pull request by xla**
To give shape usage more flexibility this changeset adds rotations which expect an angle and a center of rotation. The additional information is passed via the vertexes and used in the vertex shader with a rotation matrix.

* add Rotation to Line and Rectangle
* expose zero Rotation
* implement clock example show-casing rotations

### CODE REVIEW

File: examples/shapes_simple.rs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+    let window = Window::new(&event_loop).unwrap();
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Setup renderer
+    ///////////////////////////////////////////////////////////////////////////
+
+    let mut r = Renderer::new(&window);
+    let mut win = window.inner_size();
+
+    println!("{:?}", window.inner_size());
```
Comment (cloudhead) [inline examples/shapes_simple.rs]: :x:

File: src/kit/data/shape.vert [hunk 2]
```diff
@@ -10,12 +10,37 @@ layout(set = 1, binding = 0) uniform Model {
...
 
 layout(location = 0) in vec2 position;
-layout(location = 1) in vec4 color;
+layout(location = 1) in float angle;
+layout(location = 2) in vec2 center;
+layout(location = 3) in vec4 color;
 
 layout(location = 0) out vec4 f_color;
 
+mat4 rotationZ(float angle ) {
```
Comment (cloudhead) [inline src/kit/data/shape.vert]:
```suggestion
mat4 rotationZ(float angle) {
```

Comment (cloudhead) [reply src/kit/data/shape.vert]: I think you can remove this function.

File: src/kit/data/shape.vert [hunk 3]
```diff
@@ -10,12 +10,37 @@ layout(set = 1, binding = 0) uniform Model {
...
+	return mat4(
+		cos(angle), -sin(angle),  0, 0,
+		sin(angle),  cos(angle),  0, 0,
+		0,          0,            1, 0,
+		0,          0,            0, 1
+	);
+}
+
+mat2 rotate2d(float a) {
```
Comment (cloudhead) [inline src/kit/data/shape.vert]:
```suggestion
mat2 rotate2d(float angle) {
```

Comment (cloudhead) [reply src/kit/data/shape.vert]: Maybe rename function to `rotation2d`, cause it's confusing with the other `rotate` function.

File: src/kit/shape2d.rs [hunk 4]
```diff
@@ -140,33 +153,52 @@ pub enum Fill {
...
+        Self {
+            angle: 0.0,
+            center: Point2 { x: 0.0, y: 0.0 },
+        }
+    }
+
+    pub fn new(angle: f32, center: Point2<f32>) -> Self {
+        Self { angle, center }
+    }
+}
```
Comment (cloudhead) [inline src/kit/shape2d.rs]: Would be nice to have a `Default` instance also, which is basically `zero`

**Comment by xla** [comment]
@cloudhead Thanks for the quick review, believe all you points are addressed.

File: examples/clock.rs [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    let window = Window::new(&event_loop).unwrap();
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Setup renderer
+    ///////////////////////////////////////////////////////////////////////////
+
+    let mut r = Renderer::new(&window);
+    let mut win = window.inner_size();
+
+    println!("{:?}", window.inner_size());
```
Comment (cloudhead) [inline examples/clock.rs]: This still ^

**Pull request merged**